File: 202-HappyNumber/202-HappyNumber.cpp
// Last updated: 4/4/2026, 11:05:58 PM
class Solution {
public:
    bool isHappy(int n) {
        int s=n;
        int f=n;
        do{
            s=findsquare(s);
            f=findsquare(findsquare(f));
        }while(s!=f);
        return (s==1);
    }
    int findsquare(int n){
        int sum=0;
        while(n!=0){
            int rem=n%10;
            sum+=(rem*rem);
            n/=10;
        }
        return sum;
    }
};

File: 202-HappyNumber/202-HappyNumber.java
// Last updated: 4/4/2026, 11:06:04 PM
class Solution {
    public boolean isHappy(int n) {
        
        while(n>9){
            int sum=0;
            int digit=0;
            while(n!=0){
                digit=n%10;
                sum+=(digit*digit);
                n=n/10;
            }
             n=sum;
        }
           
        
        if(n==1 || n==7){
            return true;
        }else{
            return false;
        }
}
};
